### ISSUE TITLE
Package patterns not working as intended

### ISSUE THREAD

**Issue by mgotovac** [skuzzle/restrict-imports-enforcer-rule#177]
I'm using package pattern, specifically "`**.DumbName` would match every import which ends in `DumbName`", but it's not applied properly.

Plugin configuration:
```
<restrictImports implementation="org.apache.maven.plugins.enforcer.RestrictImports">
    <basePackages>
        <basePackage>com.console.sbi.rest.model.*</basePackage>
    </basePackages>
    <bannedImport>com.console.model.**.Entity</bannedImport>
</restrictImports>
```

In a class `com.console.sbi.rest.model.User` I have an import `com.console.model.UserEntity` and the rule is **not** triggered. If I replace the pattern with "com.console.model.**" all works as expected.

Plugin version: 2.5.0

**Comment by skuzzle** [comment]
Hi, 

Thx for your feedback. If I understand correctly, you try to match a substring of a classname with your pattern. You want to forbid to import any type which name ends in `...Entity`. Unfortunately that is not something that is currently supported. 
The only workaround for now is the one you already posted. Namely, forbidding the whole sub package.

I can see how the documentation is a bit misleading here.

**Comment by skuzzle** [comment]
I've also just spiked the possibility of allowing a wildcard character within a package part to match prefixes and suffixes. I'll need to give this some more thoughts and testing, but maybe I'll come up with a solution for your problem

**Comment by mgotovac** [comment]
That would be greatly appreciated! Thanks!

**Comment by skuzzle** [comment]
I've just released 2.6.0. You could now use
```
    <bannedImport>com.console.model.*Entity</bannedImport>
```
for your use-case.

**Comment by mgotovac** [comment]
Tried it and it works as intended! Thanks for your help
